Backend/app/dashboard/Lib/src/Domain/Action/Dashboard/Person.hs

```diff
@@ -251,9 +253,13 @@ assignRole ::
   m APISuccess
 assignRole _ personId roleId = do
   _person <- QP.findById personId >>= fromMaybeM (PersonDoesNotExist personId.getId)
-  _role <- QRole.findById roleId >>= fromMaybeM (RoleDoesNotExist roleId.getId)
+  role <- QRole.findById roleId >>= fromMaybeM (RoleDoesNotExist roleId.getId)
   QP.updatePersonRole personId roleId
-  pure Success
+  let assignRoleWithPersonId = UIP.assignRole (cast personId)
```

Comment:
Wouldn't work this way @Shamillion 
You need client call instead of direct using of bpp function (see other apis for example)